common/src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -509,6 +510,33 @@ class SchemaService(private val options: Neo4jOptions, private val driverCache:
       .asBoolean()
   }
 
+  def validateQuery(query: String, expectedQueryTypes: org.neo4j.driver.summary.QueryType*): String = try {
+    val queryType = session.run(s"EXPLAIN $query").consume().queryType()
+    if (expectedQueryTypes.isEmpty || expectedQueryTypes.contains(queryType)) {
+      ""
+    } else {
+      s"Query non valid because the expected type should be [${expectedQueryTypes.mkString(", ")}], but the actual type is $queryType"
+    }
+  } catch {
+    case e: Throwable => s"Query not compiled for the following exception: ${ExceptionUtils.getMessage(e)}"
+  }
+
+  def validateQueryCount(query: String): String = try {
+    val resultSummary = session.run(s"EXPLAIN $query").consume()
+    val queryType = resultSummary.queryType()
+    val plan = resultSummary.plan()
+    val expectedQueryTypes = Set(org.neo4j.driver.summary.QueryType.READ_ONLY, org.neo4j.driver.summary.QueryType.SCHEMA_WRITE)
+    val isReadOnly = expectedQueryTypes.contains(queryType)
+    val hasCountIdentifier = plan.identifiers().asScala.toSet == Set("count")
+    if (isReadOnly && hasCountIdentifier) {
+      ""
+    } else {
+      s"Query count non valid because the expected type should be [${expectedQueryTypes.mkString(", ")}], but the actual type is $queryType"
```

Comment:
same question as above

Reply:
In this case the query cames from the compiler itself so it's already included. Please check the tests